src/components/checkbox/daikin-checkbox.ts

```diff
@@ -145,85 +121,78 @@ export class DaikinCheckbox extends LitElement {
   label = "";
 
   /**
-   * Specify the component size
+   * Specify the label position.
+   * - `right` (default): The label will be placed to the right of the checkbox.
+   * - `left`: The label will be placed to the left of the checkbox.
+   * - `hidden`: The label will not be shown.
    */
-  @property({ type: String })
-  size: CheckboxVariantProps["size"] = "small";
+  @property({ type: String, attribute: "label-position" })
+  labelPosition: CheckboxVariantProps["labelPosition"] = "right";
 
   /**
-   * Specify the label position
-   * when `left` the label will be in left of checkbox, when `right` label will be in right of checkbox
+   * Specify whether the checkbox is be checked
    */
-  @property({ type: String, attribute: "label-position" })
-  labelPosition: "left" | "right" = "right";
+  @property({ type: String, reflect: true, attribute: "check-state" })
+  checkState: "unchecked" | "indeterminate" | "checked" = "unchecked";
 
   /**
    * Specify whether the Checkbox should be disabled
    */
   @property({ type: Boolean, reflect: true })
   disabled = false;
 
-  /**
-   * Specify whether the checkbox is read only
-   */
-  @property({ type: Boolean, reflect: true })
-  readonly = false;
+  get checked() {
+    return this.checkState === "checked";
+  }
```

Comment:
Maybe we should have a setter as well.

```suggestion
  get checked(): boolean {
    return this.checkState === "checked";
  }

  set checked(value: boolean): void {
    this.checkState = value ? "checked" : "unchecked";
  }
```